spec/app_support/reservations/duration_change_validations_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+
+describe Reservations::DurationChangeValidations do
+
+  subject(:validator) { described_class.new(reservation) }
+  let(:reservation) { create :setup_reservation }
+
+  describe "#start_time_not_changed" do
+    context "with an upcoming reservation" do
+      before do
+        reservation.assign_attributes(
+          reserve_start_at: 1.minute.from_now,
+          reserve_end_at: 61.minutes.from_now)
+      end
+
+      it { expect(validator).to be_valid }
+    end
+
+    context "with an ongoing reseration" do
+      before do
+        reservation.assign_attributes(
+          reserve_start_at: 30.minutes.ago,
+          reserve_end_at: 30.minutes.from_now)
+      end
+
+      it { expect(validator).to be_invalid }
```

Comment:
It's probably worth checking that the `errors` contains the right thing. That way we're not returning false positives when some other validation is causing it to be invalid.